lib/mongoid/history/trackable.rb

```diff
@@ -294,80 +299,76 @@ def track_history_for_action(action)
+        def embeds_one?(field)
+          relation_of(field) == Mongoid::Relations::Embedded::One
         end
 
         # Indicates whether there is an Embedded::Many relation for the given embedded field.
         #
-        # @param [ String | Symbol ] embed The name of the embedded field
+        # @param [ String | Symbol ] field The name of the embedded field.
         #
-        # @return [ Boolean ] true if there is an Embedded::Many relation for the given embedded field
-        def embeds_many?(embed)
-          relation_of(embed) == Mongoid::Relations::Embedded::Many
+        # @return [ Boolean ] true if there is an Embedded::Many relation for the given embedded field.
+        def embeds_many?(field)
+          relation_of(field) == Mongoid::Relations::Embedded::Many
         end
 
-        # Return the class for embeds_many relation
+        # Retrieves the database representation of an embedded field name, in case the :store_as option is used.
         #
-        # @param [ String ] field The database field name for embedded relation
+        # @param [ String | Symbol ] embed The name or alias of the embedded field.
         #
-        # @return [ nil | Constant ]
-        def embeds_many_class(field)
-          @embeds_many_class ||= {}
-          return @embeds_many_class[field] if @embeds_many_class.key?(field)
-          field_alias = aliased_fields.key(field)
-          relation = relations
-                     .select { |_, v| v.relation == Mongoid::Relations::Embedded::Many }
-                     .detect { |rel_k, _| rel_k == field_alias }
-          @embeds_many_class[field] = relation && relation.last.class_name.constantize
+        # @return [ String ] The database name of the embedded field.
+        def relation_alias(embed)
+          relation_aliases[embed]
         end
 
-        # Retrieves the database representation of an embedded field name, in case the :store_as option is used.
+        protected
+
+        # Return the reflected metadata for a relation.
         #
-        # @param [ String | Symbol ] embed The name or alias of the embedded field
+        # @param [ String ] field The database field name for a relation.
         #
-        # @return [ String ] the database name of the embedded field
-        def embedded_alias(embed)
-          embedded_aliases[embed]
+        # @return [ nil | Mongoid::Relations::Metadata ]
+        def meta_of(field)
+          @meta_of ||= {}
+          return @meta_of[field] if @meta_of.key?(field)
+          @meta_of[field] = reflect_on_association(relation_alias(field))
         end
 
-        protected
+        # Returns a relation for the given field.
+        #
+        # @param [ String | Symbol ] field The name of the field.
+        #
+        # @return [ nil | Constant ] Type of relation.
+        def relation_of(field)
+          meta = meta_of(field)
+          meta ? meta.relation : nil
```

Comment:
Since we have a dependency on activesupport, would it be safe to use `meta.try(:relation)` here?

Reply:
It would, but it would hide any exception such as the method `.relation` no longer exists.

Reply:
Good point. Too bad we can't rely on Ruby 2.3 safe navigation operator `&.` yet.